### ISSUE TITLE
Error: Process exited with code null

### ISSUE THREAD

**Issue by MichaelC001** [gildas-lormeau/single-file-cli#91]
![CleanShot_2024_05_27_201331](https://github.com/gildas-lormeau/single-file-cli/assets/90915827/a020e1f8-71ec-4e81-b582-e66700bbd38a)
MacOS gets the error when I attempt to save the page

**Comment by gildas-lormeau** [comment]
This is due to the fact the Docker image is built for x86 processors.

**Comment by mumoing** [comment]
Had the same problem. window x86
I use the following command
`npm install "single-file-cli"`
`cd node_modules/single-file-cli`
Then the code will keep stuck and no results. like this:
![image](https://github.com/gildas-lormeau/single-file-cli/assets/35980988/d6fef3b4-080e-4e9a-8a14-016cc3987727)
I used the release version `single-file.exe` and the result was the same.

---

When I go to the `.bin` directory and execute it, I get a correct `.html file`, but the terminal will report an error like this.
![image](https://github.com/gildas-lormeau/single-file-cli/assets/35980988/16bf5371-4aaa-46c3-8e3b-649d6a6bdf2c)

My Deno version is as follows：
![image](https://github.com/gildas-lormeau/single-file-cli/assets/35980988/1dbf4a44-4f4b-419d-9a74-cca6bf52c7b8)

**Comment by gildas-lormeau** [comment]
I'll see if I can reproduce this problem in Windows. There's something that puzzles me though. The code is executed in Node.js instead of Deno. This can be seen in your screenshot of the displayed error. Have you renamed `single-file-node.js` to `single-file`?

**Comment by TommyBloom** [comment]
I'm getting the same error. havent been able to figure it out yet

**Comment by andjar** [comment]
Got the same deno error when using the npm installed version, but it works with the newest single-file.exe version

**Comment by gildas-lormeau** [comment]
@andjar Thank you for the additional info. This tends to confirm that the issue is related to the Node.js support.

**Comment by andjar** [comment]
My pleasure; the complete error message is:

```bash
C:\Users\andjar>single-file https://wikipedia.org wikipedia.html
Process exited with code null
file:///C:/Users/andjar/AppData/Roaming/npm/node_modules/single-file-cli/lib/deno-polyfill.js:86
                                                reject(new Error(`Process exited with code ${code}`));
                                                       ^

Error: Process exited with code null
    at ChildProcess.<anonymous> (file:///C:/Users/andjar/AppData/Roaming/npm/node_modules/single-file-cli/lib/deno-polyfill.js:86:14)
    at ChildProcess.emit (node:events:518:28)
    at ChildProcess._handle.onexit (node:internal/child_process:294:12)

Node.js v20.12.0
```

**Comment by gildas-lormeau** [comment]
@andjar Thank you, if it's possible for you, could you try to edit the file `/lib/deno-polyfill.js` in the folder where `single-file-cli` is installed, replace the line 83 (see below) and tell me if it fixes the issue when using Node.js?

Replace:
```js
if (code === 0 || code === 143 || code === 130) {
```

with:
```js
if (code === 0 || code === 143 || code === 130 || code === null) {
```

**Comment by andjar** [comment]
So, after running `winget install DenoLand.Deno` in cmd, `single-file` started working from the command line. When called from R, however, using `system()` it still failed with the same error as above. However, after adding ` || code === null` as requested, it started working from R as well.

To confirm, I removed ` || code === null` again from `/lib/deno-polyfill.js`, and the error re-appeared when called from R.

Thanks, I think you got it!

**Comment by gildas-lormeau** [comment]
Thank you for the test, I have fixed the issue in the version `2.0.45` I've just released and published on NPM.
